src/notebook/manager.ts

```diff
@@ -297,6 +309,99 @@ class NotebookManager {
   }
 
   /**
+   * Select the next cell.
+   */
+  selectNext(): void {
+    if (this.model.activeCellIndex === this.model.cells.length - 1) {
+      return;
+    }
+    this.model.activeCellIndex += 1;
+    this.deselectCells();
+  }
+
+  /**
+   * Select the previous cell.
+   */
+  selectPrev(): void {
+    if (this.model.activeCellIndex === 0) {
+      return;
+    }
+    this.model.activeCellIndex -= 1;
+    this.deselectCells();
+  }
+
+  /**
+   * Extend the selection to the previous cell.
+   */
+  extendSelectionAbove(): void {
+    let model = this.model;
+    let cells = model.cells;
+    // Do not wrap around.
+    if (model.activeCellIndex === 0) {
+      return;
+    }
+    let current = cells.get(model.activeCellIndex);
+    model.select(current);
+    model.activeCellIndex -= 1;
+    let prev = cells.get(model.activeCellIndex);
+    if (model.isSelected(prev)) {
+      model.deselect(current);
+      if (model.activeCellIndex >= 1) {
```

Comment:
To me, it's more understandable to switch the if statement test to be `model.activeCellIndex===0`, i.e., if you're the first cell, then do this, otherwise, check the cell before you. Otherwise I had to realize that the else condition meant I was the first cell. If you feel strongly the other way, that's fine too.